bundles/org.openhab.binding.neohub/src/main/java/org/openhab/binding/neohub/internal/NeoHubHandler.java

```diff
@@ -279,4 +391,73 @@ private void fastPollingSchedulerExecute() {
             lazyPollingSchedulerExecute();
         }
     }
+
+    /*
+     * select whether to use the old "deprecated" API or the new API
+     */
+    private void selectApi() {
+        boolean supportsLegacyApi = false;
+        boolean supportsFutureApi = false;
+
+        NeoHubSocket socket = this.socket;
+        if (socket != null) {
+            String responseJson;
+            NeoHubReadDcbResponse systemData;
+
+            try {
+                responseJson = socket.sendMessage(CMD_CODE_READ_DCB);
+                systemData = NeoHubReadDcbResponse.createSystemData(responseJson);
+                supportsLegacyApi = systemData instanceof NeoHubReadDcbResponse;
+            } catch (Exception e) {
+                // we learned that this API is not currently supported; no big deal
+            }
+            try {
+                responseJson = socket.sendMessage(CMD_CODE_GET_SYSTEM);
+                systemData = NeoHubReadDcbResponse.createSystemData(responseJson);
+                supportsFutureApi = systemData instanceof NeoHubReadDcbResponse;
+            } catch (Exception e) {
+                // we learned that this API is not currently supported; no big deal
+            }
+        }
+
+        if (!supportsLegacyApi && !supportsFutureApi) {
+            logger.warn("Currently neither legacy nor new API are supported!");
+            isApiOnline = false;
```

Comment:
If the api is offline shouldn't that cause the thing to go offline as well?

Reply:
It does in fact cause the thing status to go offline -- this happens when the next poll calls `fromNeoHubGetDeviceData()` which fails with an exception that is caught on lines 268..270 (note: I intentionally don't change thing status in selectApi() to avoid unnecessarily doing the same thing in several places)